### ISSUE TITLE
Permissions Error When Upgrading to Meteor 1.4.1

### ISSUE THREAD

**Issue by bsbechtel** [fourseven/meteor-scss#224]
I am trying to upgrade a Meteor app from version 1.2 to the latest stable version - 1.4.1. I keep getting the following error regarding permissions:

`While loading package fourseven:scss@3.8.1:
error: Command failed:
/Users/bsbechtel/.meteor/packages/meteor-tool/.1.4.1_1.10bdqiz++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/bin/npm
rebuild --no-bin-links --update-binary
{ [Error: EACCES: permission denied, mkdir
'/Users/bsbechtel/.meteor/packages/fourseven_scss/.3.8.1.9616xb++os.osx.x86_64+web.browser+web.cordova/plugin.compileScssBatch.os.osx.x86_64/npm/node_modules/meteor/compileScssBatch/node_modules/.temp-eq58qh/node_modules/node-sass/vendor/darwin-x64-46']
errno: -13,
code: 'EACCES',
syscall: 'mkdir',
path:
'/Users/bsbechtel/.meteor/packages/fourseven_scss/.3.8.1.9616xb++os.osx.x86_64+web.browser+web.cordova/plugin.compileScssBatch.os.osx.x86_64/npm/node_modules/meteor/compileScssBatch/node_modules/.temp-eq58qh/node_modules/node-sass/vendor/darwin-x64-46'
}
gyp info it worked if it ends with ok
gyp verb cli [
'/Users/bsbechtel/.meteor/packages/meteor-tool/.1.4.1_1.10bdqiz++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/bin/node',
gyp verb cli
'/Users/bsbechtel/.meteor/packages/fourseven_scss/.3.8.1.9616xb++os.osx.x86_64+web.browser+web.cordova/plugin.compileScssBatch.os.osx.x86_64/npm/node_modules/meteor/compileScssBatch/node_modules/node-gyp/bin/node-gyp.js',
gyp verb cli   'rebuild',
gyp verb cli   '--verbose',
gyp verb cli   '--libsass_ext=',
gyp verb cli   '--libsass_cflags=',
gyp verb cli   '--libsass_ldflags=',
gyp verb cli   '--libsass_library=' ]
gyp info using node-gyp@3.3.1
gyp info using node@4.5.0 | darwin | x64
gyp verb command rebuild []
gyp verb command clean []
gyp verb clean removing "build" directory
gyp verb command configure []
gyp verb check python checking for Python executable "python2" in the PATH
gyp verb `which` succeeded python2 /usr/local/bin/python2
gyp verb check python version `/usr/local/bin/python2 -c "import platform;
print(platform.python_version());"` returned: "2.7.9\n"
gyp verb get node dir no --target version specified, falling back to host node
version: 4.5.0
gyp verb command install [ '4.5.0' ]
gyp verb install input version string "4.5.0"
gyp verb install installing version: 4.5.0
gyp verb install --ensure was passed, so won't reinstall if already installed
gyp verb install version is already installed, need to check "installVersion"
gyp verb got "installVersion" 9
gyp verb needs "installVersion" 9
gyp verb install version is good
gyp verb get node dir target node version installed: 4.5.0
gyp verb build dir attempting to create "build" dir:
/Users/bsbechtel/.meteor/packages/fourseven_scss/.3.8.1.9616xb++os.osx.x86_64+web.browser+web.cordova/plugin.compileScssBatch.os.osx.x86_64/npm/node_modules/meteor/compileScssBatch/node_modules/.temp-eq58qh/node_modules/node-sass/build
gyp ERR! configure error 
gyp ERR! stack Error: EACCES: permission denied, mkdir
'/Users/bsbechtel/.meteor/packages/fourseven_scss/.3.8.1.9616xb++os.osx.x86_64+web.browser+web.cordova/plugin.compileScssBatch.os.osx.x86_64/npm/node_modules/meteor/compileScssBatch/node_modules/.temp-eq58qh/node_modules/node-sass/build'
gyp ERR! stack     at Error (native)
gyp ERR! System Darwin 15.6.0
gyp ERR! command
"/Users/bsbechtel/.meteor/packages/meteor-tool/.1.4.1_1.10bdqiz++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/bin/node"
"/Users/bsbechtel/.meteor/packages/fourseven_scss/.3.8.1.9616xb++os.osx.x86_64+web.browser+web.cordova/plugin.compileScssBatch.os.osx.x86_64/npm/node_modules/meteor/compileScssBatch/node_modules/node-gyp/bin/node-gyp.js"
"rebuild" "--verbose" "--libsass_ext=" "--libsass_cflags=" "--libsass_ldflags="
"--libsass_library="
gyp ERR! cwd
/Users/bsbechtel/.meteor/packages/fourseven_scss/.3.8.1.9616xb++os.osx.x86_64+web.browser+web.cordova/plugin.compileScssBatch.os.osx.x86_64/npm/node_modules/meteor/compileScssBatch/node_modules/.temp-eq58qh/node_modules/node-sass
gyp ERR! node -v v4.5.0
gyp ERR! node-gyp -v v3.3.1
gyp ERR! not ok 
Build failed
`

Note above the error is with package version 3.8.1, but I've tried it with 3.9.0 as well and keep getting the same issue. I've tried changing permissions on the package folders per the error and used sudo to install, and am still not having any luck. Has anyone else experienced this?

**Comment by WimAtIHomer** [comment]
Same problem here.

**Comment by bsbechtel** [comment]
@WimAtIHomer I ended up deleting my entire meteor install and doing a fresh install per this [thread](https://github.com/meteor/meteor/issues/7529). It wasn't really ideal, but it fixed the problem.

**Comment by bsbechtel** [comment]
So I was having this issue with my development directory, but when I switched over to my testing directory on my development machine, I ran into this same issue. I tried deleting meteor and reinstalling without any luck this time. I don't know if this is an SCSS issue or Meteor issue.

**Comment by WimAtIHomer** [comment]
I have downloaded the package to the /packages folder in my project to make it a 'local' package and then it installed fine. That is for now a workable solution for me.

**Comment by bsbechtel** [comment]
I uninstalled node-gyp and reinstalled without sudo, after changing my npm installer's permissions. This has gotten it to work in my testing directory, but I'm hesitant to close this out yet. Permissions issues with building files between Meteor, NPM, and node-gyp seems to be a big rat's nest. I don't know if @fourseven could weigh in on this one?

**Comment by Unforgiven-wanda** [comment]
Same issue as well

**Comment by bsbechtel** [comment]
So I finally got this figured out after several days of deleting, reinstalling, and changing permissions on my node, npm, and meteor installs. My first advice if you are facing this issue would be to make sure your node/npm install doesn't require sudo to run (see [this thread](https://github.com/sass/node-sass/issues/1098)). The same goes for Meteor, although installing node via sudo seems to be more common. Once that is out of the way, the issue you are probably facing has to do with cached or previous builds within your project. The easiest way I figured out how to track down those issues is to just run your project with meteor, and then `sudo chown -Rv $USER:$GROUP /directory/giving/EACCESS/error`. Hopefully this helps!
